src/syntax/parser.rs

```diff
@@ -76,44 +73,44 @@ impl<TokType: Tokenizer + Iterator<Item=Tokens>> Parser<TokType> {
                             if self.indent_level == depth {
                                 self.block_status = BlockStatus::In;
                             } else {
-                                return self.write_error("Invalid level of indentation");
+                                return Some(self.write_error("Invalid level of indentation"));
                             }
                         },
                         BlockStatus::In => {
                             self.block_status = BlockStatus::Out;
                         },
                     }
                     if !self.between_brackets && allow_any {
-                        return result;
+                        return token;
                     }
                 },
-                Comment(_) => {
+                Some(Comment(_)) => {
                     self.last_depth = None;
                 },
-                Error(_) => {
-                    // self.write_error(&err);
-                    return result;
-                },
-                _ => {
-                    return result;
-                },
+                _ => return token,
             }
         }
     }
 
     fn next_token(&mut self) -> Tokens {
-        self._next_token(false)
+        if let Some(token) = self._next_token(false) {
```

Comment:
this part is kind of unfortunate since it repeats for each method, but my hope is this will go away anyway